<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mcg::View</class>
 <widget class="QMainWindow" name="mcg::View">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>802</width>
    <height>561</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MazeCaveGenerator</string>
  </property>
  <widget class="QWidget" name="central_widget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="20,1">
    <item>
     <widget class="Loader" name="mcg_loader" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tab_widget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>271</width>
        <height>0</height>
       </size>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">QTabWidget::tab-bar {
                alignment: center;
            }</string>
      </property>
      <property name="tabPosition">
       <enum>QTabWidget::North</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="elideMode">
       <enum>Qt::ElideNone</enum>
      </property>
      <widget class="QWidget" name="tab_cave">
       <attribute name="title">
        <string>Cave</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="3" column="0">
         <layout class="QGridLayout" name="grid_layout_cave">
          <item row="6" column="0" colspan="2">
           <widget class="QLabel" name="label_born_limit">
            <property name="text">
             <string>Born limit</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QSpinBox" name="spinbox_cave_rows"/>
          </item>
          <item row="8" column="1">
           <widget class="QPushButton" name="pushbutton_stop_autogen_cave">
            <property name="text">
             <string>Stop</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0">
           <widget class="QLabel" name="label_delay">
            <property name="text">
             <string>Delay, ms</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="spinbox_life_chance"/>
          </item>
          <item row="4" column="0" colspan="2">
           <widget class="QLabel" name="label_live_limit">
            <property name="text">
             <string>Live limit</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QSpinBox" name="spinbox_born_limit_max"/>
          </item>
          <item row="5" column="1">
           <widget class="QSpinBox" name="spinbox_live_limit_max"/>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_life_chance">
            <property name="text">
             <string>Life chance</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QSpinBox" name="spinbox_live_limit_min"/>
          </item>
          <item row="2" column="1">
           <widget class="QSlider" name="horizontal_slider_cave_cols">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="9" column="1">
           <widget class="QSpinBox" name="spinbox_delay">
            <property name="minimum">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QSlider" name="horizontal_slider_cave_rows">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label_cave_cols">
            <property name="text">
             <string>Cols</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QPushButton" name="pushbutton_autogen_cave">
            <property name="text">
             <string>Autogeneration</string>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QSpinBox" name="spinbox_born_limit_min"/>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="spinbox_cave_cols"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_cave_rows">
            <property name="text">
             <string>Rows</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="pushbutton_open_cave">
          <property name="text">
           <string>Open</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="pushbutton_save_cave">
          <property name="text">
           <string>Save</string>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <spacer name="vertical_spacer_cave">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="4" column="0">
         <widget class="QPushButton" name="pushbutton_generate_cave_next">
          <property name="text">
           <string>Next generation</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QPushButton" name="pushbutton_draw_mode_cave">
          <property name="text">
           <string>DrawMode</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="pushbutton_generate_cave">
          <property name="text">
           <string>Generate</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_maze">
       <attribute name="title">
        <string>Maze</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="4" column="0">
         <spacer name="vertical_spacer_maze">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="pushbutton_save_maze">
          <property name="text">
           <string>Save</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="pushbutton_open_maze">
          <property name="text">
           <string>Open</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="pushbutton_generate_maze">
          <property name="text">
           <string>Generate</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <layout class="QGridLayout" name="grid_layout_maze">
          <item row="1" column="0">
           <widget class="QSpinBox" name="spinbox_maze_rows"/>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="spinbox_maze_cols"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_maze_rows">
            <property name="text">
             <string>Rows</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label_maze_cols">
            <property name="text">
             <string>Cols</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QSlider" name="horizontal_slider_maze_rows">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSlider" name="horizontal_slider_maze_cols">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Loader</class>
   <extends>QWidget</extends>
   <header>view/loader.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
